doc/build/sysbuild/images.rst

```diff
@@ -0,0 +1,361 @@
+
+.. code-block:: cmake
+
+   set_config_bool(${DEFAULT_IMAGE} CONFIG_BUILD_OUTPUT_HEX y)
+
+These can safely be used in an application/board/SoC ``sysbuild.cmake`` file, as that file is
+included before the image CMake processes are invoked. When extending
+:ref:`sysbuild using modules <sysbuild_module_integration>` the pre-CMake hook should be used
+instead, for example:
+
+.. code-block:: cmake
+
+   function(${SYSBUILD_CURRENT_MODULE_NAME}_pre_cmake)
+     cmake_parse_arguments(PRE_CMAKE "" "" "IMAGES" ${ARGN})
+
+     foreach(image ${PRE_CMAKE_IMAGES})
+       set_config_bool(${image} CONFIG_BUILD_OUTPUT_HEX y)
+     endforeach()
+   endfunction()
+
+Image configuration script
+--------------------------
+
+An image configuration script is a CMake file that can be used to configure an image with common
+configuration values, multiple can be used per image, the configuration should be transferrable to
+different images to correctly configure them based upon options set in sysbuild. MCUboot
+configuration options are configured in both the application the MCUboot image using this method
+which allows sysbuild to be the central location for things like the signing key which are then
+kept in-sync in the main application bootloader images.
+
+Image configuration has an allow-list of names which must be set in the Zephyr file
+:zephyr_file:`share/sysbuild/cmake/modules/sysbuild_extensions.cmake` in the
+``ExternalZephyrProject_Add`` function. After adding a new type, it can be used when adding a
+sysbuild image, for example:
+
+.. tabs::
+
+   .. group-tab:: ``sysbuild_extensions.cmake``
+
+      Full file path: ``share/sysbuild/cmake/modules/sysbuild_extensions.cmake``
+
+      .. code-block:: cmake
+
+         # ...
+         # Usage:
+         #   ExternalZephyrProject_Add(APPLICATION <name>
+         #                             SOURCE_DIR <dir>
+         #                             [BOARD <board> [BOARD_REVISION <revision>]]
+         #                             [APP_TYPE <MAIN|BOOTLOADER|MY_CUSTOM_TYPE>]
+         #   )
+         # ...
+         # APP_TYPE <MAIN|BOOTLOADER|MY_CUSTOM_TYPE>: Application type.
+         #                                            MAIN indicates this application is the main application
+         #                                            and where user defined settings should be passed on as-is
+         #                                            except for multi image build flags.
+         #                                            For example, -DCONF_FILES=<files> will be passed on to the
+         #                                            MAIN_APP unmodified.
+         #                                            BOOTLOADER indicates this app is a bootloader
+         #                                            MY_CUSTOM_TYPE indicates this app is...
+         # ...
+         function(ExternalZephyrProject_Add)
+           set(app_types MAIN BOOTLOADER MY_CUSTOM_TYPE)
```

Comment:
Should we also highlight the `IMAGE_CONF_SCRIPT` target property? Ref: https://github.com/zephyrproject-rtos/zephyr/pull/59665#discussion_r1241976701
In fact, it might make more sense to document that instead of how to modify sysbuild internals.

Reply:
Have added a new section and moved this one to the bottom